### PR TITLE
Apr 1356 task list view model tech debt

### PR DESCRIPTION
**Pull request by David-B-Read**

### CODE REVIEW

File: src/SFA.DAS.ApplyService.Web.UnitTests/Controllers/RoatpApplicationControllerTests.cs [hunk 1]
```diff
@@ -84,14 +88,20 @@ public void Before_each_test()
...
                                                          _processPageFlowService.Object, _pagesWithSectionsFlowService.Object,
                                                          _questionPropertyTokeniser.Object, _pageOverrideConfiguration.Object,
                                                          _pageNavigationTrackingService.Object, _qnaLinks.Object, _customValidatorFactory.Object,
-                                                         _notRequiredOverrides.Object, _roatpApiClient.Object,
-                                                         _submitApplicationEmailService.Object, _tabularDataRepository.Object)
+                                                         _roatpApiClient.Object,
+                                                         _submitApplicationEmailService.Object, _tabularDataRepository.Object,
+                                                         _roatpTaskListWorkflowService.Object, _roatpOrganisationVerificationService.Object,
+                                                         _notRequiredOverridesService.Object, _taskListWorkflowService.Object)
```
Comment (MarkFCain) [inline src/SFA.DAS.ApplyService.Web.UnitTests/Controllers/RoatpApplicationControllerTests.cs:104]: wow look at all those dependencies...

File: src/SFA.DAS.ApplyService.Web.UnitTests/Services/RoatpTaskListWorkflowServiceTests.cs [hunk 2]
```diff
@@ -1,29 +1,53 @@
...
+            _service = new RoatpTaskListWorkflowService(_qnaApiClient.Object, _notRequiredOverridesService.Object, config.Object, logger.Object);
+        }
+
         [Test]
         public void Get_SectionStatus_Empty_When_Null_Sequences_Used()
         {
             var expectedResult = string.Empty;
-            var actualResult = RoatpTaskListWorkflowService.SectionStatus(null, null, 0, 0);
+            var actualResult = _service.SectionStatus(_applicationId, 0, 0, null, null);
             Assert.AreEqual(expectedResult, actualResult);
```
Comment (MarkFCain) [inline src/SFA.DAS.ApplyService.Web.UnitTests/Services/RoatpTaskListWorkflowServiceTests.cs:43]: Would be a tiny bit easier to read these tests if magic numbers became named variables.  Same for all tests in this file

File: src/SFA.DAS.ApplyService.Web/Services/RoatpOrganisationVerificationService.cs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+using SFA.DAS.ApplyService.Session;
+using SFA.DAS.ApplyService.Web.Infrastructure;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.ApplyService.Web.Services
+{
+    public class RoatpOrganisationVerificationService : IRoatpOrganisationVerificationService
```
Comment (MarkFCain) [inline src/SFA.DAS.ApplyService.Web/Services/RoatpOrganisationVerificationService.cs:13]: Couldn't see any specific tests for this. Do they exist?

File: src/SFA.DAS.ApplyService.Web/Services/NotRequiredOverridesService.cs [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+using Newtonsoft.Json.Linq;
+using SFA.DAS.ApplyService.Session;
+using SFA.DAS.ApplyService.Web.Configuration;
+using SFA.DAS.ApplyService.Web.Infrastructure;
+using System;
+using System.Collections.Generic;
+
+namespace SFA.DAS.ApplyService.Web.Services
+{
+    public class NotRequiredOverridesService : INotRequiredOverridesService
```
Comment (MarkFCain) [inline src/SFA.DAS.ApplyService.Web/Services/NotRequiredOverridesService.cs:11]: Can't see any specific tests for this. Are they present?

File: src/SFA.DAS.ApplyService.Web.UnitTests/Services/RoatpTaskListWorkflowServiceTests.cs (hunk 2 above)
Comment (David-B-Read) [reply src/SFA.DAS.ApplyService.Web.UnitTests/Services/RoatpTaskListWorkflowServiceTests.cs]: Reformatted unit tests so easier to read

File: src/SFA.DAS.ApplyService.Web/Services/RoatpOrganisationVerificationService.cs (hunk 3 above)
Comment (David-B-Read) [reply src/SFA.DAS.ApplyService.Web/Services/RoatpOrganisationVerificationService.cs:10]: Added tests for this service

File: src/SFA.DAS.ApplyService.Web/Services/NotRequiredOverridesService.cs (hunk 4 above)
Comment (David-B-Read) [reply src/SFA.DAS.ApplyService.Web/Services/NotRequiredOverridesService.cs:11]: Added tests for this service

File: src/SFA.DAS.ApplyService.Web/Services/RoatpOrganisationVerificationService.cs [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+            }
+
+            var status = new OrganisationVerificationStatus
+            {
+                VerifiedCompaniesHouse = (verifiedCompaniesHouse != null && verifiedCompaniesHouse.Value == "TRUE"),
+                VerifiedCharityCommission = (verifiedCharityCommission != null && verifiedCharityCommission.Value == "TRUE"),
+                CompaniesHouseManualEntry = (companiesHouseManualEntry != null && companiesHouseManualEntry.Value == "TRUE"),
+                CharityCommissionManualEntry = (charityCommissionManualEntry != null && charityCommissionManualEntry.Value == "TRUE"),
+                CompaniesHouseDataConfirmed = (companiesHouseDataConfirmed != null && companiesHouseDataConfirmed.Value == "Y"),
+                CharityCommissionDataConfirmed = (charityCommissionDataConfirmed != null && charityCommissionDataConfirmed.Value == "Y"),
```
Comment (martin-hornsby) [inline src/SFA.DAS.ApplyService.Web/Services/RoatpOrganisationVerificationService.cs:64]: Just looking at this... wouldn't it have been better to use consistency and have them all TRUE / FALSE rather then the last two being Y and N ?

**Pull request merged**